cstar_ocean/component.py

```diff
@@ -118,30 +117,33 @@ def __str__(self):
             f"\n{NID} InputDataset objects (query using ROMSComponent.input_datasets"
         )
 
-        base_str += "\n\nDiscretization info:"
+        disc_str=''
         if hasattr(self, "time_step") and self.time_step is not None:
-            base_str += "\ntime_step: " + str(self.time_step)
+            disc_str += "\ntime_step: " + str(self.time_step)
         if hasattr(self, "n_procs_x") and self.n_procs_x is not None:
-            base_str += (
+            disc_str += (
                 "\nn_procs_x: "
                 + str(self.n_procs_x)
                 + " (Number of x-direction processors)"
             )
         if hasattr(self, "n_procs_y") and self.n_procs_y is not None:
-            base_str += (
+            disc_str += (
                 "\nn_procs_y:"
                 + str(self.n_procs_y)
                 + " (Number of y-direction processors)"
             )
         if hasattr(self, "n_levels") and self.n_levels is not None:
-            base_str += "\nn_levels:" + str(self.n_levels)
+            disc_str += "\nn_levels:" + str(self.n_levels)
         if hasattr(self, "nx") and self.nx is not None:
-            base_str += "\nnx:" + str(self.nx)
+            disc_str += "\nnx:" + str(self.nx)
         if hasattr(self, "ny") and self.ny is not None:
-            base_str += "\nny:" + str(self.ny)
+            disc_str += "\nny:" + str(self.ny)
         if hasattr(self, "exe_path") and self.exe_path is not None:
-            base_str += "\n\nIs compiled: True"
-            base_str += "\n exe_path: " + self.exe_path
+            disc_str += "\n\nIs compiled: True"
+            disc_str += "\n exe_path: " + self.exe_path
+        if len(disc_str)>0:
```

Comment:
Regardless of how the inputs are constructed (https://github.com/CWorthy-ocean/C-Star/issues/15) this should be a separate class internally, so that this becomes something like

```python
return base_str + str(self.discretization)
```

If this works though we can deal with that in a separate PR.

Reply:
Agree - the restructuring here isn't in response to #15 but just for consistency in the case where no discretization info is present.